### PR TITLE
27 backend mvp

### PR DESCRIPTION
**Pull request by zjcers**
Fixes issue #11 #27 

## List of Changes

 - Added a flask_restful + SQLAlchemy API skeleton
 - Implemented a ping endpoint and a provisioning endpoint
 - Added a test suite
 - Wrote basic documentation

### CODE REVIEW

**Review by aamirani — APPROVED** [review]

File: radiomanager/radiomanager/api/endpoints/handshaking.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    POST_PARSER = reqparse.RequestParser()
+    POST_PARSER.add_argument("name", type=str)
+    POST_PARSER.add_argument("public_key", type=str)
+    ALPHANUMERIC_WITH_SPACES = re.compile("([a-z]|[A-Z]|[0-9]| )*")
+
+    def post(self):
+        """
+        POSTing to this endpoint provisions a new client with no authentication
+        capabilities. The client is expected to pass a UUID.
+        """
```
Comment (jackdweber) [inline radiomanager/radiomanager/api/endpoints/handshaking.py:53]: Curious on how UUID works, how will the device always generate a truly unique ID without knowing the other radios?

**Review by jackdweber — COMMENTED** [review]
So for uuid we will generate it on the radio via uuid#() of some sort. Will this always get us the same uuid on different boots, that way we don't lose configurations? Also what about using like the pi serial # as some sort of hard coded id, possibly allowing us to have a manufacturing database that a console could use to get information on, report stolen etc. That may be thinking to far out of scope too.

File: radiomanager/radiomanager/api/endpoints/handshaking.py (hunk 1 above)
Comment (zjcers) [reply radiomanager/radiomanager/api/endpoints/handshaking.py:53]: The UUID bit should be removed. The server should assign the client some sort of unique identifier upon a successful provision.

Comment (zjcers) [reply radiomanager/radiomanager/api/endpoints/handshaking.py:53]: I've decided to actually implement the UUID field as well as enforce that it is unique. In practice, UUIDs are basically collision free (key space is too big), but there's a protection (with a test) included anyway.

**Comment by zjcers** [comment]
@jackdweber I couldn't find a good place to get a unique hardware id from, so I went down the UUID route. The client will have to persist that as part of its state files.

**Review by jackdweber — APPROVED** [review]
I think the changes here were also looked at, be me, in the 31 PR.

**Pull request merged**